ddtrace/opentelemetry/tracer_test.go

```diff
@@ -85,6 +89,74 @@ func TestTracerOptions(t *testing.T) {
 	assert.Contains(fmt.Sprint(sp), "dd.env=wrapper_env")
 }
 
+func TestSpanContext(t *testing.T) {
+	assert := assert.New(t)
+	tp := NewTracerProvider()
+	defer tp.Shutdown()
+	otel.SetTracerProvider(tp)
+	tr := otel.Tracer("")
+
+	pCtx, parent := tr.Start(context.Background(), "parent")
+	pSpanCtx := parent.SpanContext()
+
+	_, child := tr.Start(pCtx, "child")
+	cSpanCtx := child.SpanContext()
+
+	assert.Equal(cSpanCtx.TraceFlags(), pSpanCtx.TraceFlags())
+	assert.Equal(cSpanCtx.TraceID(), pSpanCtx.TraceID())
+	assert.Equal(cSpanCtx.IsRemote(), pSpanCtx.IsRemote())
+	assert.Equal(cSpanCtx.TraceState().String(), pSpanCtx.TraceState().String())
+}
+
+func TestForceFlush(t *testing.T) {
+	assert := assert.New(t)
+	testData := []struct {
+		timeOut time.Duration
+		flushed bool
+	}{
+		{timeOut: 0 * time.Second, flushed: false},
+		{timeOut: 300 * time.Second, flushed: true},
+	}
+	success := false
+	setFlushFail := func(ok bool) { success = ok }
+
+	for _, tc := range testData {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		tp, payloads, cleanup := mockTracerProvider(t)
+		tr := otel.Tracer("")
+		_, sp := tr.Start(context.Background(), "test_span")
+		sp.End()
+		tp.ForceFlush(tc.timeOut, setFlushFail)
+		payload, err := waitForPayload(ctx, payloads)
+		if tc.flushed {
+			if err != nil {
+				t.Log(err)
+				t.Fail()
+			}
+			assert.Contains(payload, "test_span")
+		}
+		assert.Equal(tc.flushed, success)
+		success = false
+		cleanup()
+		cancel()
+	}
+}
+
+func TestShutdown(t *testing.T) {
+	assert := assert.New(t)
+	tp := NewTracerProvider()
+	otel.SetTracerProvider(tp)
+
+	testLog := new(log.RecordLogger)
+	defer log.UseLogger(testLog)()
+
+	tp.Shutdown()
+	tp.ForceFlush(5*time.Second, func(ok bool) {})
+
+	logs := testLog.Logs()
+	assert.Contains(logs[len(logs)-1], "Cannot perform (*TracerProvider).Flush since the tracer is already stopped")
```

Comment:
I'd really like to avoid checking warning messages, since it'll be a pain to change this test if we want to re-word the warning. Is there another way we can ensure that it shut down correctly?